### PR TITLE
Fix ifdef for shadow inconsistency bug fix

### PR DESCRIPTION
**Pull request by lyzhan7**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [x] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

Fix regression for ifdef for shadow inconsistency bug fix (checked in https://github.com/microsoft/react-native-macos/commit/a598ee4eff9364f5e8865de092f7d368388eb032)

We were seeing a red box in iOS (copied below), after this fix confirmed that this box is gone.
MicrosoftTeams-image (2)

## Changelog

[macOS] [Fixed] - Mixed up if defs for shadow inconsistency bug

### CODE REVIEW

File: React/Views/RCTViewManager.m [hunk 1]
```diff
@@ -197,7 +197,7 @@ - (RCTShadowView *)shadowView
 RCT_REMAP_VIEW_PROPERTY(opacity, alphaValue, CGFloat)
 #endif // ]TODO(macOS GH#774)
 
-#if TARGET_OS_OSX // TODO(macOS GH#774)
+#if !TARGET_OS_OSX // TODO(macOS GH#774)
```
Comment (harrieshin) [inline React/Views/RCTViewManager.m:200]: how did this work before? and nothing failed?

**Review by harrieshin — APPROVED** [review]

**Comment by harrieshin** [comment]
please add description which commit has cause the regression

**Comment by harrieshin** [comment]
please add vlaidation notes

File: React/Views/RCTViewManager.m (hunk 1 above)
Comment (lyzhan7) [reply React/Views/RCTViewManager.m:200]: Yeah I had the same question, I didn't test on IOS when I made this change but I feel like I should have seen errors when I tested on macOS

Comment (Saadnajmi) [reply React/Views/RCTViewManager.m:200]: Perhaps a copy-paste error since you tested in FURN and not RN-Tester?

**Review by chiuam — APPROVED** [review]

**Comment by lyzhan7** [comment]
The Apple PR pipeline is failing with this error: 

"RNTesterLoadAllPages
  AnExApp, RedBox errors: Error setting property 'shadowRadius' of RCTText with tag #13: Exception thrown while executing UI block: -[RCTTextView setShadowRadius:]: unrecognized selector sent to instance 0x617000940a80 (NSInternalInconsistencyException)". 

This seems like an actual issue with the original PR, I think the mistake with ifdefs hid the issue and somehow the original PR passed these tests. This will need to be looked at more, so going to close this PR for now and revert the PR that introduced this until we find a fix.

**Pull request closed without merging**